GDJS/Runtime/AsyncTasksManager.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * GDevelop JS Platform
+ * Copyright 2013-2022 Florian Rival (Florian.Rival@gmail.com). All rights reserved.
+ * This project is released under the MIT License.
+ */
+namespace gdjs {
+  /**
+   * A manager for all asynchronous tasks.
+   * @see {@link RuntimeScene.getAsyncTasksManager}.
+   */
+  export class AsyncTasksManager {
+    /**
+     * Maps a task to the callback to be executed once it is finished.
+     */
+    private tasks = new Array<{
+      asyncTask: AsyncTask;
+      callback: (runtimeScene: gdjs.RuntimeScene) => void;
+    }>();
+
+    /**
+     * Run all pending asynchronous tasks.
+     */
+    processTasks(runtimeScene: RuntimeScene): void {
+      for (const task of this.tasks) {
+        if (task.asyncTask.update(runtimeScene)) {
+          // The task has finished, run the callback and remove it.
+          task.callback(runtimeScene);
+          this.tasks.splice(this.tasks.indexOf(task), 1);
```

Comment:
To avoid this `indexOf`, you could use an index to browser `this.tasks`. It seems more efficient to me to have a possibly useless number variable, than having potentially lots of `indexOf` to run.

Reply:
Isn't this action of splicing the array you're looping on a bit dangerous?

Reply:
I used the index of because I was worried of the splicing changing the indices, but I realize in hindsight that it doesn't make sense. I'll need to experiment to see what happens when slicing while iterating. This does look error prone indeed, thanks for spotting that.

Reply:
Replaced the automatic iterator with a manual for loop, there indeed was a bug that skipped over the iteration over one task when removing one before it 👍